conan/tools/cmake/cmakedeps2/target_configuration.py

```diff
@@ -269,6 +271,29 @@ def _get_exes(self, cpp_info, pkg_name, pkg_folder, pkg_folder_var):
 
         return exes
 
+    def _get_sources(self, cpp_info, pkg_name, pkg_folder, pkg_folder_var):
+        sources = {}
+
+        if cpp_info.has_components:
+            for name, comp in cpp_info.components.items():
+                if comp.sources:
+                    target_name = self._cmakedeps.get_property("cmake_target_name", self._conanfile,
+                                                               name)
+                    target = target_name or f"{pkg_name}::{name}"
+                    sources[target] = []
+                    for source in comp.sources:
+                        sources[target].append(self._path(source, pkg_folder, pkg_folder_var))
+                    sources[target] = ";".join(sources[target])
```

Comment:
Maybe the string concatenation of CMake with ``;`` should happen in the template, and the sources provide a dict{target: [list of files]}